test/libyul/ControlFlowGraphTest.cpp

```diff
@@ -0,0 +1,207 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <test/libyul/ControlFlowGraphTest.h>
+#include <test/libyul/Common.h>
+#include <test/Common.h>
+
+#include <libyul/backends/evm/ControlFlowGraph.h>
+#include <libyul/backends/evm/ControlFlowGraphBuilder.h>
+#include <libyul/Object.h>
+#include <liblangutil/SourceReferenceFormatter.h>
+
+#include <libsolutil/AnsiColorized.h>
+#include <libsolutil/Visitor.h>
+
+using namespace solidity;
+using namespace solidity::util;
+using namespace solidity::langutil;
+using namespace solidity::yul;
+using namespace solidity::yul::test;
+using namespace solidity::frontend;
+using namespace solidity::frontend::test;
+using namespace std;
+
+ControlFlowGraphTest::ControlFlowGraphTest(string const& _filename):
+	TestCase(_filename)
+{
+	m_source = m_reader.source();
+	auto dialectName = m_reader.stringSetting("dialect", "evm");
+	m_dialect = &dialect(dialectName, solidity::test::CommonOptions::get().evmVersion());
+	m_expectation = m_reader.simpleExpectations();
+}
+
+namespace
+{
+static std::string stackSlotToString(StackSlot const& _slot)
+{
+	return std::visit(util::GenericVisitor{
+		[](FunctionCallReturnLabelSlot const& _ret) -> std::string { return "RET[" + _ret.call.get().functionName.name.str() + "]"; },
+		[](FunctionReturnLabelSlot const&) -> std::string { return "RET"; },
+		[](VariableSlot const& _var) { return _var.variable.get().name.str(); },
+		[](LiteralSlot const& _lit) { return util::toCompactHexWithPrefix(_lit.value); },
+		[](TemporarySlot const& _tmp) -> std::string { return "TMP[" + _tmp.call.get().functionName.name.str() + ", " + std::to_string(_tmp.index) + "]"; },
+		[](JunkSlot const&) -> std::string { return "JUNK"; }
```

Comment:
No, it will only happen with the ``StackLayoutGenerator`` when merging control flow later on.
Example:
```
let x := 42
// Stack layout: [x]
if calldataload(0) {
  mstore(x, 0) // last use of x
}
// Since the "false"-branch directly jumps here, I do have ``x`` on stack. but it's never used again, so I the "entry"-layout
// of the post-if-block is ``[ JUNK ]``. This gives the exit of the "true"-branch the freedom to produce any value
// on the stack before jumping here instead of requiring it to keep a copy of ``x`` that's not needed.
```